R/boxplot.R

```diff
@@ -6,9 +6,93 @@
 #' @param facet Determines whether separate graphs will be created for each category
 #'
 #' @return a ggplot object
+#'
+#'
+#'
 #' @export
 #'
 #' @examples
 #' function(df, Length, Species, TRUE)
-boxplot <- function(df, x, y, facet = FALSE) {
+#'
+boxplot <- function(df, x, y, facet = FALSE){
+
+library(viridis)
+library(stringr)
+library(rlang)
+library(ggplot2)
+
+### input tests
+#check type of x
+if ( typeof(rlang::get_expr(ggplot2::vars({{ x }})[[1]])) !=
+     typeof(as.list(quote(symbol))[[1]])){
+    stop("x, the column name for x, must be unquoted")
+}
+
+#check type of y
+if ( typeof(rlang::get_expr(ggplot2::vars({{ y }})[[1]])) !=
+         typeof(as.list(quote(symbol))[[1]])){
+        stop("y, the column name of y, must be unquoted")
+}
+
+#check type of df
+if ( typeof(df) != 'list' ){
+        stop("df must be a dataframe")
+}
+
+#check if x exists in df
+if ( !(as.character(as.list(
+    rlang::get_expr(ggplot2::vars({{ x }})[[1]])
+    )))
+    %in%
+    colnames({{ df }})
+){
+    stop('x column name not found')
+}
+
+#check if y exists in df
+if ( !(as.character(as.list(
+        rlang::get_expr(ggplot2::vars({{ y }})[[1]])
+    )))
+    %in%
+    colnames({{ df }})
+){
+    stop('y column name not found')
+}
+
```

Comment:
You should also add tests to check if the data type of x is a "character"/"factor" and that of y is "numeric"/"float"/"int" in the dataframe.

Reply:
Good suggestion!